modules/sdk-coin-avaxp/src/lib/transactionBuilder.ts

```diff
@@ -148,6 +156,172 @@ export abstract class TransactionBuilder extends BaseTransactionBuilder {
+        const outputidx = utils.cb58Decode(utxo.outputidx);
+        const addressesIndex = utxo.addressesIndex ?? [];
+
+        // either user (0) or recovery (2)
+        const firstIndex = this.recoverSigner ? 2 : 0;
+        const bitgoIndex = 1;
+        currentTotal = currentTotal.add(amt);
+
+        const secpTransferInput = new SECPTransferInput(amt);
+
+        if (!buildOutputs) {
+          addressesIndex.forEach((i) => secpTransferInput.addSignatureIdx(i, this.transaction._fromAddresses[i]));
+        } else {
+          // if user/backup > bitgo
+          if (addressesIndex[bitgoIndex] < addressesIndex[firstIndex]) {
+            secpTransferInput.addSignatureIdx(addressesIndex[bitgoIndex], this.transaction._fromAddresses[bitgoIndex]);
+            secpTransferInput.addSignatureIdx(addressesIndex[firstIndex], this.transaction._fromAddresses[firstIndex]);
+            credentials.push(
+              SelectCredentialClass(
+                secpTransferInput.getCredentialID(), // 9
+                ['', this.transaction._fromAddresses[firstIndex].toString('hex')].map(utils.createSig)
+              )
+            );
+          } else {
+            secpTransferInput.addSignatureIdx(addressesIndex[firstIndex], this.transaction._fromAddresses[firstIndex]);
+            secpTransferInput.addSignatureIdx(addressesIndex[bitgoIndex], this.transaction._fromAddresses[bitgoIndex]);
+            credentials.push(
+              SelectCredentialClass(
+                secpTransferInput.getCredentialID(),
+                [this.transaction._fromAddresses[firstIndex].toString('hex'), ''].map(utils.createSig)
+              )
+            );
+          }
+        }
+
+        const input: TransferableInput = new TransferableInput(
+          txidBuf,
+          outputidx,
+          this.transaction._assetId,
+          secpTransferInput
+        );
+        inputs.push(input);
+      }
+    });
+
+    if (buildOutputs) {
+      if (currentTotal.lt(totalTarget)) {
+        throw new BuildTransactionError(
+          `Utxo outputs get ${currentTotal.toString()} and ${totalTarget.toString()} is required`
+        );
+      } else if (currentTotal.gt(totalTarget)) {
+        outputs.push(
+          new TransferableOutput(
+            this.transaction._assetId,
+            new SECPTransferOutput(
+              currentTotal.sub(totalTarget),
+              this.transaction._fromAddresses,
+              this.transaction._locktime,
+              this.transaction._threshold
+            )
+          )
+        );
+      }
+    }
+    // get outputs and credentials from the deserialized transaction if we are in OVC
+    return {
+      inputs,
+      outputs: outputs.length === 0 ? this.transaction.avaxPTransaction.getOuts() : outputs,
+      credentials: credentials.length === 0 ? this.transaction.credentials : credentials,
+    };
+  }
+
+  /**
+   * Inputs can be controlled but outputs get reordered in transactions
```

Comment:
does the comment match what it's doing? 
`this.transaction._utxos = this.recoverUtxos(baseTx.getImportInputs());` 
doesnt this mean input's not controlled?